WordPress/src/main/java/org/wordpress/android/ui/posts/PromoDialog.java

```diff
@@ -53,19 +79,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        Button btn = (Button) view.findViewById(R.id.promo_dialog_button_positive);
-        btn.setText(mButtonPositiveId);
         ImageView image = (ImageView) view.findViewById(R.id.promo_dialog_image);
         image.setImageResource(mDrawableId);
         WPTextView title = (WPTextView) view.findViewById(R.id.promo_dialog_title);
         title.setText(mTitleId);
         WPTextView desc = (WPTextView) view.findViewById(R.id.promo_dialog_description);
         desc.setText(mDescriptionId);
-        btn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                getDialog().cancel();
-            }
-        });
+
+        Button btn = (Button) view.findViewById(R.id.promo_dialog_button_positive);
+        btn.setText(mButtonPositiveId);
+        if (mPositiveButtonOnClickListener == null) {
+            btn.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    getDialog().cancel();
+                }
+            });
+        } else {
+            btn.setOnClickListener(mPositiveButtonOnClickListener);
```

Comment:
We're lacking a dialog `cancel()` call here like we do in https://github.com/wordpress-mobile/WordPress-Android/pull/6467/files#diff-1999fe3664a86fade02c8922f6857c75R132 

The default click handler does issue a cancel:
https://github.com/wordpress-mobile/WordPress-Android/pull/6467/files#diff-42c777438a1a186bb67bd763a279089eR91
I’m wondering, shouldn’t we make sure that when you set a handler we _also_  cancel the dialog, either by giving the new handler that responsibility, or if not, we could maybe set our own anonymous handler and call the other listener within it, like this:
```
            btn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    getDialog().cancel();
                    mPositiveButtonOnClickListener.onClick(view);
                }
            });
```